serde_arrow/src/internal/schema.rs

```diff
@@ -580,6 +586,19 @@ impl std::str::FromStr for GenericDataType {
             };
 
             Ok(GenericDataType::Timestamp(unit, Some(s.to_string())))
+        } else if let Some(s) = s.strip_prefix("Time(") {
+            let unit = if s.strip_prefix("Second, ").is_some() {
```

Comment:
`s.strip_prefix().is_some()` => `s.starts_with`. 

But I would expect that this overall logic is inccorrect (hard to say without tests).  E.g., your Display impl woudl generate `Time(Nanosecond))`, but this will not be parsed by this logic..